### PR TITLE
Encapsulate params in a Parameters class

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@dobe please review

File: sql/src/main/java/io/crate/analyze/Parameters.java [hunk 1]
```diff
@@ -22,50 +22,47 @@
...
-    private final List<?> params;
+    private static final Object[] EMPTY_ROW = new Object[0];
+    public static final Row EMPTY = new Row() {
+        @Override
+        public int size() {
+            return 0;
+        }
+
+        @Override
+        public Object get(int index) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/Parameters.java]: this should fail always

File: sql/src/main/java/io/crate/analyze/Parameters.java [hunk 2]
```diff
@@ -22,50 +22,47 @@
...
 import java.util.Collections;
 import java.util.List;
 
 public class Parameters {
 
-    public static final Parameters EMPTY = new Parameters(Collections.emptyList());
-    public static final List<Parameters> EMPTY_BULK = Collections.emptyList();
-    private final List<?> params;
+    private static final Object[] EMPTY_ROW = new Object[0];
+    public static final Row EMPTY = new Row() {
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/Parameters.java]: would it make sense to put this into core.collections ?

Comment (dobe) [inline sql/src/main/java/io/crate/analyze/Parameters.java]: maybe a Rows util class? ... might also contain some constructor functions ...

File: sql/src/main/java/io/crate/analyze/Parameters.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import io.crate.core.collections.Row;
+import io.crate.core.collections.RowN;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Parameters {
+
+    private static final Object[] EMPTY_ROW = new Object[0];
```
Comment (dobe) [inline sql/src/main/java/io/crate/analyze/Parameters.java]: EMPTY_CELLS would match more the existing naming

File: core/src/main/java/io/crate/core/collections/Row.java [hunk 4]
```diff
@@ -25,6 +25,27 @@
...
+        private final Object[] EMPTY_CELLS = new Object[0];
+
+        @Override
+        public int size() {
+            return 0;
+        }
+
+        @Override
+        public Object get(int index) {
+            throw new IllegalArgumentException("EMPTY row has no cells");
```
Comment (dobe) [inline core/src/main/java/io/crate/core/collections/Row.java]: use indexoutofboundsex

**Comment by dobe** [comment]
otherwise ack

**Pull request merged**